### PR TITLE
feat: implement comprehensive API key authentication with full Unicode support

### PR DESCRIPTION
**Pull request by anstrom**
## Summary

This PR implements API key authentication with full Unicode support for international users.

### Authentication System
- **API Key Generation**: Secure keys with  prefix and base32 encoding
- **bcrypt Hashing**: With SHA-256 pre-hashing for keys >72 bytes
- **Unicode Names**: Full international character support (Chinese, Arabic, Japanese, Russian, Emoji)
- **Security Validation**: Blocks control characters while allowing printable Unicode
- **Role-Based Access Control**: Foundation for future permission systems

### CLI Commands
- `scanorama auth create-key --name 'Production Key 🔑'`
- `scanorama auth list-keys --active`
- `scanorama auth show-key <id>`
- `scanorama auth update-key <id>`
- `scanorama auth revoke-key <id>`

### API Integration
- **Bearer Token Auth**: `Authorization: Bearer sk_abc123...`
- **Middleware**: Secure validation with user context injection
- **Error Handling**: Proper 401/403 responses with helpful messages

## 🌍 Unicode Support

**Supported Characters:**
- ✅ International scripts (中文, العربية, 日本語, Русский, 한글)
- ✅ Emoji and symbols (🔑, ✨, ∑, ∆, ∇)
- ✅ Accented characters (café, naïve, résumé)
- ✅ Mathematical symbols and special punctuation

**Security Measures:**
- ❌ Blocks ASCII control characters (0-31, 127)
- ❌ Blocks Unicode control ranges (C1, bidirectional, isolates)
- ❌ Prevents injection attacks via malicious Unicode

## 📚 Documentation

- **API Documentation**: 219 lines covering authentication flows
- **Validation Rules**: 604 lines detailing Unicode handling and validation
- **Examples**: JavaScript, Python, cURL authentication patterns
- **Error Codes**: Detailed validation error responses

## 🧪 Testing

**Test Coverage:**
- ✅ 15 Unicode test cases (multilingual, edge cases)
- ✅ bcrypt handling for all key lengths
- ✅ Security validation (control character blocking)
- ✅ Display prefix consistency
- ✅ All authentication flows

**Performance:**
- ✅ 0 linting issues (fixed nestif complexity)
- ✅ Handles 1000+ character keys efficiently
- ✅ No hanging tests (fixed RTL override validation)

## 🗄️ Database

- **Migration 006**: API keys table with proper indexing
- **UUID Primary Keys**: For secure, non-enumerable IDs
- **Expiration Support**: Optional key expiration dates
- **Usage Tracking**: Last used timestamps and usage counts
- **RBAC Ready**: Junction tables for future role assignments

## Breaking Changes

- **Authentication**: New middleware affects all API endpoints (can be disabled)
- **Database**: Requires migration 006 for API key storage

## Migration Guide

1. Run database migrations: `make migrate`
2. Create your first key: `scanorama auth create-key --name 'Admin Key'`
3. Update API clients to include `Authorization: Bearer <key>` header
4. Test Unicode names: `scanorama auth create-key --name 'Test 测试 🔑'`

## Performance Impact

- **Minimal overhead**: bcrypt validation ~50ms per request
- **Memory efficient**: SHA-256 pre-hashing for long keys
- **Database indexed**: Fast key lookups with proper indexes

---

**Ready for production use with enterprise-grade security and international support.**

### CODE REVIEW

File: cmd/cli/apikeys.go [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+	keyInfo, err := storeAPIKey(database, generatedKey)
+	if err != nil {
+		return fmt.Errorf("failed to store API key: %w", err)
+	}
+
+	fmt.Println("🔑 API Key Created Successfully")
+	fmt.Println("═══════════════════════════════")
+	fmt.Printf("Key ID: %s\n", keyInfo.ID)
+	fmt.Printf("Name: %s\n", keyInfo.Name)
+	fmt.Printf("Prefix: %s\n", keyInfo.KeyPrefix)
```
[the remark this answers is not in the document]
Comment (anstrom) [reply cmd/cli/apikeys.go:416]: the user needs to get the cleartext key
